### PR TITLE
resolves !63

### PR DESCRIPTION
**Pull request by ryno75**
…rom sts called identity for host account/session

### CODE REVIEW

**Comment by ryno75** [comment]
I stopped short of editing the CHANGELOG.

**Comment by ryno75** [comment]
Resolves issue !63

**Comment by iainelder** [comment]
@ryno75 , thanks for the PR ! I'll have a look this week. 

Github is supposed to run the tests automatically in CI, but I can't see the output here right now. I don't know if that's because it's hidden from me or because it hasn't run for some reason. (@connelldave , has something changed in the CI config?)

You don't need to update the changelog. That's normally done after one or more PRs are merged and just before a new release is published.

[Boto 3 doesn't try to explain what a partition is](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.get_available_partitions), so I don't think we should either :-) For convenience we could link to the [ARN format](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) that explains what a partition is.

Curiously, the boto3 SDK lists more available partitions than are documented there!

```console
$ python -c 'import boto3; print(boto3.Session().get_available_partitions())'
['aws', 'aws-cn', 'aws-us-gov', 'aws-iso', 'aws-iso-b']
```

**Comment by connelldave** [comment]
> (@connelldave , has something changed in the CI config?)

No - to avoid being able to hijack Actions, they only run for commits directly to this repo by maintainers. Otherwise someone could fork and cat out the deploy token for pypi :)

**Comment by ryno75** [comment]
> @ryno75 , thanks for the PR ! I'll have a look this week.
> 
> Github is supposed to run the tests automatically in CI, but I can't see the output here right now. I don't know if that's because it's hidden from me or because it hasn't run for some reason. (@connelldave , has something changed in the CI config?)
> 
> You don't need to update the changelog. That's normally done after one or more PRs are merged and just before a new release is published.
> 
> [Boto 3 doesn't try to explain what a partition is](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.get_available_partitions), so I don't think we should either :-) For convenience we could link to the [ARN format](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) that explains what a partition is.
> 
> Curiously, the boto3 SDK lists more available partitions than are documented there!
> 
> ```
> $ pip freeze | grep '^boto3=='
> boto3==1.26.86
> $ python -c 'import boto3; print(boto3.Session().get_available_partitions())'
> ['aws', 'aws-cn', 'aws-us-gov', 'aws-iso', 'aws-iso-b']
> ```

Thanks for looking @iainelder !
Shall I go ahead and drop [these lines](https://github.com/connelldave/botocove/pull/64/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R231-R239) from the `README.md`?

File: README.md
Comment (iainelder) [inline README.md:238]:
```suggestion
If not provided, Cove will use the [AWS partition](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html)
of your profile in constructing the ARN for the role to assume in all target
accounts.
```

**Comment by iainelder** [comment]
@ryno75, when I run the tests locally, I get some errors. We'll need to fix these so that we can merge this change.

Now I see that the README lacks a "developer" or "contributor" quick start guide! So here is how you run the tests locally:

Install [Poetry](https://python-poetry.org/).

Run the following commands in the repo:

```bash
poetry install
poetry run pre-commit run --all-files
```

I see errors from mypy and pytest, typical errors when adding a new parameter because some of the tests expect a certain shape of output or input and the new parameter changes those shapes.

Do you want to have a go at fixing these errors?

**Comment by iainelder** [comment]
Similar PRs that may offer guidance:

* https://github.com/connelldave/botocove/pull/10
* https://github.com/connelldave/botocove/pull/21

**Comment by ryno75** [comment]
TIL: `poetry`... NICE!
Updated and pushed.

**Comment by ryno75** [comment]
oh, my eyes dost fail me. More updates pending...

**Comment by ryno75** [comment]
ok, all tests passing now

**Comment by iainelder** [comment]
@ryno75 , thanks. I'll have a look this week.

Yeah, Poetry is great :-)

**Comment by iainelder** [comment]
@ryno75 All the tests pass here. The code changes look clean. Good work!

There are no new tests and that's fine by me.

We use the [moto](https://github.com/getmoto/moto) mocking framework for faking AWS interactions. I'm unsure whether it simulates GovCloud. [Some issues](https://github.com/search?q=repo%3Agetmoto%2Fmoto+govcloud&type=issues) mention it but no documentation. 

In future we could add robust validation and error handling of the `partition` input. See #42 and #49 for examples of tests respectively against the `region` input and the `target_ids` input. 

The new documentation doesn't say what happens when `partition` is actually specified. If you are expecting to set the value explcitly, then we should probably document how that behaves.

I'm guessing you need this because you are running from the management account of a GovCloud organization and you want botocove to access the member accounts of the same GovCloud organization.

(Today I learned that each GovCloud  account has its own [linked standard account](https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/getting-started-standard-account-linking.html) and that the GovCloud organization exists as a distinct resource from the standard organization.)

I have no access to GovCloud, so I might struggle to imagine the use cases. I'll need to trust what you tell me about it :-)

**Comment by iainelder** [comment]
@connelldave, how does it look?

LGTM from my side.

**Comment by connelldave** [comment]
> @connelldave, how does it look?
> 
> LGTM from my side.

Yeah, makes sense to me - makes me a little itchy we're composing an ARN using so many args, but I can't think of any particularly good alternatives that justify changing the existing function signature. I'll cut a release - thank you for contributing @ryno75!

**Review by connelldave — APPROVED** [review]

**Pull request merged**
